### PR TITLE
Log a warning when using default settings for live data

### LINKED ISSUE #33649: Log warning when using default settings for reflectometry live data
Currently in the ISIS reflectometry interface you have the option to start running live data reduction. This uses settings from the wildcard row in the table on the Experiment Settings tab, or algorithm defaults, even if the run angle matches a specific row in the lookup table. Implementing correct lookup is tricky as described in #33641. For now we should show a warning in the log when starting the live data monitor that explains which defaults will be used.

### PR DESCRIPTION
**Pull request by gemmaguest**
**Report to:** Max at ISIS

**Description of work.**

This PR adds warnings if potentially unexpected settings are used when running the live data monitor from the ISIS Reflectometry interface.

I've added warnings both in the RunsPresenter when requesting the defaults, as well as in createAlgorithmRuntimeProps when it searches for the wildcard row. The former allows us to warn specifically about live data, and the latter to say specifically which defaults are used. Although this is slight duplication I think it is better to be over-informative here because it is very confusing if unexpected defaults get used. The latter function is not used anywhere else at the moment but the warnings will still make sense if it is in the future.

The warnings will appear consecutively and are written such that they make sense together and look like a single message:
image

**To test:**

*Live data*

- Open the ISIS Reflectometry interface
- Click Start Monitor
- This warning about using a wildcard row should appear in the log:
```
Note that lookup of experiment settings by angle/title is not supported for live data.
Using experiment settings from the wildcard row.
```
- On the Experiment Settings tab, click on the empty row and press Delete. The table should now be empty
- Back on the Runs tab click Stop Monitor (if necessary) and then Start Monitor again.
- You should now get this warning:
```
Note that lookup of experiment settings by angle/title is not supported for live data.
No experiment settings found; using algorithm defaults. You may wish to specify a wildcard row in the Experiment Settings table to override them.
```

*Normal reduction*

- Set instrument to INTER and in the table enter run `13460` and angle `0.5`
- On the Experiment Settings tab make sure there is one wildcard row (i.e. angle and title are empty)
- Back on the Runs tab, click Process. You should get a warning about using a wildcard row.
- On the lookup table, change the angle to `0.5` and re-process. There should be no warning.
- On the lookup table, delete the row so the table is empty, and re-process. There should be a warning about using algorithm defaults.

Fixes #33649

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/ISISReflectometry/GUI/Batch/RowProcessingAlgorithm.cpp [hunk 1]
```diff
@@ -251,7 +253,13 @@ std::unique_ptr<MantidQt::API::IAlgorithmRuntimeProps> createAlgorithmRuntimePro
   updateExperimentProperties(*properties, model.experiment());
   updateInstrumentProperties(*properties, model.instrument());
   // Update properties from the wildcard row in the lookup table
-  updateLookupRowProperties(*properties, model.findWildcardLookupRow());
+  if (auto lookupRow = model.findWildcardLookupRow()) {
+    g_log.warning("Using experiment settings from the wildcard row.");
+    updateLookupRowProperties(*properties, *lookupRow);
+  } else {
+    g_log.warning("No experiment settings found; using algorithm defaults. You may wish to specify a wildcard"
```
Comment (ConorMFinn) [inline qt/scientific_interfaces/ISISReflectometry/GUI/Batch/RowProcessingAlgorithm.cpp:260]: Do we want these warnings to replace the ones in LookupTable.h? At the moment they are somewhat duplicated when running a regular, non-live reduction

**Review by cailafinn — CHANGES_REQUESTED** [review]
Looks good and works well, just a quick question:

File: qt/scientific_interfaces/ISISReflectometry/GUI/Batch/RowProcessingAlgorithm.cpp (hunk 1 above)
Comment (gemmaguest) [reply qt/scientific_interfaces/ISISReflectometry/GUI/Batch/RowProcessingAlgorithm.cpp]: I've updated the test instructions with some tests for a normal reduction as well as the live data tests.

**Review by cailafinn — APPROVED** [review]
Looks good to me now! 👍

**Comment by gemmaguest** [comment]
Added a patch candidate label because this behaviour is so confusing for users and they are hitting the problem right now in real experiments that this could be worth including if a patch release were to go out

**Pull request merged**